src/state.rs

```diff
@@ -509,6 +509,76 @@ impl<'a> LoadedState<'a> {
         self.persistent.save(client, storage)?;
         Ok(())
     }
+
+    /// Avoid having too many RSA keys in volatile storage
+    fn limit_cache_size(
+        client: &mut impl crate::card::Client,
+        keys: &mut [(&mut Option<KeyId>, bool)],
+    ) {
+        for key in keys
+            .iter_mut()
+            .filter_map(|(k, is_rsa)| if *is_rsa { Some(k.take()) } else { None })
+            .flatten()
+        {
+            syscall!(client.delete(key));
+        }
+    }
+
+    /// Returns the requested key
+    pub fn key_id(
+        &mut self,
+        client: &mut impl crate::card::Client,
+        key: KeyType,
+        storage: Location,
+    ) -> Result<KeyId, Status> {
+        use KeyType as K;
+        use UserVerifiedInner as V;
+        // Self::clear_cached is there to avoid having multiple keys in the volatile storage.
```

Comment:
outdated comment?

Reply:
Indeed, the method was initially called `clear_cache`, I modified the comment to `limit_cache_size`, I still think the comment is relevant.